File: MediaLoader/src/main/java/com/app/medialoader/tinyhttpd/request/Request.java
package com.app.medialoader.tinyhttpd.request;

import com.app.medialoader.tinyhttpd.HttpHeaders;
import com.app.medialoader.tinyhttpd.HttpVersion;

/**
 * 请求接口
 *
 * @author vincanyang
 */
public interface Request {

    HttpMethod method();

    String url();

    HttpVersion protocol();

    HttpHeaders headers();

    String getParam(String name);
}

File: MediaLoader/src/main/java/com/app/medialoader/tinyhttpd/request/Request.kt
package com.app.medialoader.tinyhttpd.request

import com.app.medialoader.tinyhttpd.HttpHeaders
import com.app.medialoader.tinyhttpd.HttpVersion

interface Request {
    fun method(): HttpMethod
    fun url(): String?
    fun protocol(): HttpVersion
    fun headers(): HttpHeaders
    fun getParam(name: String): String
}